doc/ooapi-v6-plan.md

```diff
@@ -0,0 +1,140 @@
+
+Dit plan is een weergave van de consensus van de verschillende stakeholders (in ieder geval de opdrachtgever en het implementatieteam). Op basis van dit plan kan een inschatting worden gemaakt van de benodigde ontwikkelcapaciteit en de verwachte doorlooptijd.
+
+# Aanpak
+
+## Architectuur en context
+
+Uitrollen nieuwe versie en uitfaseren oude versie.
+
+De overgang van OOAPI v5 naar OOAPI v6 zal een aanzienlijke tijd duren. Onderwijsinstellingen moeten hun API's bijwerken en zullen dat allemaal volgens hun eigen planning (en die van hun leveranciers) doen. Het is dus belangrijk dat we instellingen de mogelijkheid bieden om snel aan de slag te gaan met v6, maar ook om zo lang als nodig is (te bepalen door de opdrachtgever) de OOAPI v5 interfaces betrouwbaar te ondersteunen, inclusief updates voor beveiliging, aanpassingen aan externe koppelingen, enz.
+
+Voor de RIO Mapper is het daarbij relevant dat de RIO API ook op een eigen tijdpad doorontwikkeld wordt. Zolang OOAPI v5 en v6 diensten aangeboden worden, moet de RIO interface in beide versies up-to-date gehouden worden.
+
+Om dit te faciliteren, kiezen we voor het ontwikkelen en uitrollen van onafhankelijke OOAPI v5 en OOAPI v6 diensten.
+
+Van de Eduhub-gerelateerde diensten die door Jomco worden ontwikkeld, worden "standalone" OOAPI v6 implementaties opgeleverd, zodat de huidige OOAPI v5 "stack" onafhankelijk draait van de nieuwe OOAPI v6. Dit maakt het eenvoudiger om snel te itereren tijdens het ontwikkelen en testen van de nieuwe v6 stack, terwijl de huidige v5 stack stabiel blijft werken.
+
+Omdat we met deze opzet in de codebase van de RIO mapper kunnen afdwingen welke versie van de OOAPI van toepassing is, en we dus geen "als v5 - dan doe X, als v6 - dan doe Y" paden hoeven te implementeren, maakt dit het ook simpeler om de v5 code buiten gebruik te stellen op het moment dat deze niet meer gebruikt wordt.
+
+De door ons ontwikkelde diensten die in de v6 stack komen te draaien zijn de volgende:
+
+- [Eduhub Gateway](https://github.com/suRFnet/eduhub-gateway)
+
+Naar verwachting zal het volstaan om een nieuwe deploy te doen van de bestaande gateway, met aanpassingen aan de configuratie zodat deze overeenkomt met de OOAPI v6 OpenAPI specificatie.
+
+- [Registry Integration](https://github.com/SURFnet/eduhub-registry-integration)
+
+Nieuwe deploy, gekoppeld met Eduhub Gateway, geen code aanpassingen.
+
+- [RIO Mapper](https://github.com/SURFnet/eduhub-rio-mapper/)
+
+Zoals verder beschreven in dit document.
+
+- [Eduhub Validator Service](https://github.com/SURFnet/eduhub-validator-service/)
+
+Nieuwe deploy met OOAPI v6 configuratie, of OOAPI v6 als extra profiel in bestaande instantie. We verwachten weinig tot geen code aanpassingen.
+
+
+## Functionele wijzigingen in v6 Mapper
+
+In de nieuwe versie van de RIO mapper worden een aantal functionaliteiten gewijzigd of toegevoegd:
```

Comment:
Zijn dit wijzigingen die alleen voor v6 gelden of juist voor beiden?  De laatste optie lijkt mij het verstandigst.